02_bubble/sarahdavy1983/main_test.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"bytes"
+	"reflect"
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	main()
+
+	expected := strconv.Quote("[1 2 3 5]\n[1 2 3 5]\n")
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main(), expected = %v, actual = %v", expected, actual)
+	}
+}
+
+var tests = []struct {
+	name  string
+	input []int
+	want  []int
+}{
+	{"valid", []int{3, 2, 1, 5}, []int{1, 2, 3, 5}},
+	{"same number", []int{5, 5, 5, 5}, []int{5, 5, 5, 5}},
+	{"blank", []int{}, []int{}},
+	{"sorted", []int{1, 2, 3, 4}, []int{1, 2, 3, 4}},
+	{"negative number", []int{1, -5, 6, 9}, []int{-5, 1, 6, 9}},
+}
```

Comment:
I would suggest adding a test case for a 1-element slice and a 2-element slice. This is because "1" is a boundary condition for the length of the slice - you start counting at 1 and use `<`. If you get the logic slightly wrong in the code, it's easy to be off by one.

These off by one errors (boundary conditions) are common enough that you should look at the comparisons in your code and add test cases for either side of that boundary.

This is a general software engineering comment, not Go-specific, so this won't block approval if you would prefer to merge and move on. Just consider it for later labs.